qa/L0_model_update/instance_update_test.py

```diff
@@ -43,18 +43,35 @@
     set_delay,
     update_instance_group,
     update_model_file,
+    update_sequence_batching,
 )
 from tritonclient.utils import InferenceServerException
 
+# The server is not restarted across different unit test cases. If a case would
+# to fail, all subsequent cases could also fail, because the model state may not
```

Comment:
"case" do you mean the test case (`def test_xxx`)? If so, shouldn't you have implement proper setup / teardown method for the test so each test is independent from the result of the other tests?

Reply:
I added a `teardown` method to unload the model if a test case would to fail: [Add best effort to reset model state after a single test case failure](https://github.com/triton-inference-server/server/pull/5831/commits/99f59355474384719266c03686d62cb7816553be).

For simple failures, like an instance finalize count does not match, this will properly reset the model state to allow the next test case to function:
```
***
*** Failed model instance update test on rate limit mode off
***
+ cat instance_update_test.rate_limit_off.log
F...
======================================================================
FAIL: test_cpu_instance_update (__main__.TestInstanceUpdate)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/tritonserver/qa/L0_model_update/instance_update_test.py", line 186, in test_cpu_instance_update
    self.assertTrue(False, "Artificial failure")
AssertionError: False is not true : Artificial failure

----------------------------------------------------------------------
Ran 4 tests in 64.633s

FAILED (failures=1)
```
The first case failed, but the following cases passed.

If the server would to get stuck for some reasons, then this is insufficient and requires the server to be force stopped and restarted by the parent script. However, the goal of this test is to not restart the server for all the tests, so if restarting is the only fix, then it should be considered as an entire test failure.